dev/fattest.simplicity/src/componenttest/topology/utils/tck/TCKUtilities.java

```diff
@@ -865,4 +874,81 @@ static void assertServerHappy(LibertyServer server) throws Exception {
         assertThat("Server log has errors after starting", server.findStringsInLogs("CWWKF0002E"), empty()); // A bundle could not be found for ...
         assertThat("Server log has errors after starting", server.findStringsInLogs("CWWKE0702E"), empty()); // Could not resolve module ...
     }
+
+    private static Set<String> detectedShortNames = null;
+
+    private static Set<String> getDetectedShortNames() {
+        if (detectedShortNames != null) {
+            return detectedShortNames;
+        }
+
+        detectedShortNames = new HashSet<String>();
+        Bootstrap b;
+        try {
+            b = Bootstrap.getInstance();
+        } catch (Exception e) {
+            Log.info(c, "getDetectedShortNames", "Bootstrap.getInstance() failed. This will likely cause the current repeat to skip " + e.toString());
+            detectedShortNames = null; //if bootstrap is broken we've got bigger problems, but we can avoid caching an exception
+            return new HashSet<String>();
+        }
+
+        String installRoot = b.getValue("libertyInstallPath");
+        String libPath = installRoot + "/lib/features/";
+
+        PathMatcher matcher = FileSystems.getDefault().getPathMatcher("glob:*.mf");
+
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(Paths.get(libPath))) {
+            for (Path path : stream) {
+                if (!Files.isDirectory(path) && matcher.matches(path.getFileName())) {
+                    Files.lines(path)
+                                    .filter(line -> line.startsWith("IBM-ShortName: "))
+                                    .map(line -> line.replaceAll("IBM-ShortName: ", ""))
+                                    .forEach(detectedShortNames::add);
+                }
+            }
+        } catch (IOException e) {
+            Log.info(c, "getDetectedShortNames", "IO exception while reading .mf Files. This will likely cause the current repeat to skip " + e.toString());
+            detectedShortNames = null; //avoid caching an exception
+            return new HashSet<String>();
+        }
+
+        return detectedShortNames;
+
+    }
+
+    /**
+     * Checks if a feature or features are present in wlp/lib/features by reading the mf files
+     * and checking if all the provided IBM-ShortName are present.
+     *
+     * @param  featureSet A FeatureSet, all the names returned by getFeatures() will be checked for a matching IBM-ShortName in a mf file
+     * @return            true if all provided IBM-ShortName can be found in a .mf file in wlp/lib/features.
+     */
+    public static boolean areFeaturesMissing(FeatureSet featureSet) {
+        return areFeaturesMissing(featureSet.getFeatures());
+    }
```

Comment:
The method is named `areFeaturesMissing` but it returns `true` if features are _not_ missing.

I think we should renamed it to something like `allFeaturesPresent`.

Reply:
As you pointed out, it does actually return `true` if features are missing, even though the javadoc says it will return `false`.

However, as noted above, I think we should change `repeatWithPredicate` / `repeatIf` so that it does a repeat if the predicate returns `true`, in which case we want to leave the javadoc, change the name to `areAllFeaturesPresent` and reverse the logic in the method to match the javadoc.